### PR TITLE
Modified Dam project to support Data dependent routing and MultiSharduery,

### PR DESCRIPTION
**Pull request by supragyan**
Modified Dam project to support Data dependent routing and MultiSharduery,
- Implemented code to support Data dependent routing and MultiSharduery,
- Removed unnecessary code as per review comments.

### CODE REVIEW

File: root_VS2012/programs/C#/Frameworks/Infrastructure/Business/Business/MyFcBaseLogic.cs [hunk 1]
```diff
@@ -194,11 +195,30 @@ protected override void UOC_DoAction(BaseParameterValue parameterValue, ref Base
                 #region データ プロバイダ選択
 
                 // SQL Server / SQL Client用のDamを生成
-                dam = new DamSqlSvr();
```
Comment (daisukenishino) [inline root_VS2012/programs/C#/Frameworks/Infrastructure/Business/Business/MyFcBaseLogic.cs]: @supragyan 
I think this fix is not have relationship of this issue.
Please explain why it was necessary to modify.

Comment (supragyan) [inline root_VS2012/programs/C#/Frameworks/Infrastructure/Business/Business/MyFcBaseLogic.cs]: @daisukenishino 
>I think this fix is not have relationship of this issue.

Yes this fix is required for this issue.

To support Data dependent routing and MultiShardquery in Framework, [below modification](https://github.com/SymphonyTeleca/OpenTouryoTemplates/pull/166/files/946ad73440bd4bf403b25e28d095d9f752aef4b2#diff-190a334a9ff6675527c0a9623ff38c2cR198) is required.

As per [this discussion](https://github.com/SymphonyTeleca/OpenTouryoTemplates/pull/153#issuecomment-222925211) to resolve damframework-chngs branch issue, we have created new [azuretmplt-chngs branch](https://github.com/SymphonyTeleca/OpenTouryoTemplates/pull/153#discussion_r67466710) as per [this instruction](https://github.com/SymphonyTeleca/OpenTouryoTemplates/pull/153#discussion_r67100495).

Comment (daisukenishino) [inline root_VS2012/programs/C#/Frameworks/Infrastructure/Business/Business/MyFcBaseLogic.cs]: @supragyan 

**DamSqlDbWithMultiShard** requires special reference of "**Elastic database client library**".
Therefore, **MyFcBaseLogic** should not link **DamSqlDbWithMultiShard** module by default.

Therefore, **MyFcBaseLogic.UOC_ConnectionOpen()** method should be kept the states of comment-out, or **MyFcBaseLogic.UOC_ConnectionOpen()** method should get the customization of  from "damframework-chngs" branch each time.

And finally, I chose a way to get the customization of  from "damframework-chngs" branch each time.
It is mistake to fix this for **MyFcBaseLogic.UOC_ConnectionOpen()**.

**Comment by supragyan** [comment]
@daisukenishino 

>And finally, I chose a way to get the customization of from "damframework-chngs" branch each time.

We have modified as follows in this Pull request :
- Comment out Multishard and DataDependentRouting code changes in MyFCBaseLogic.cs
- Removed DamSqlDBWithMultishard project references from AllComponent.sln and Business.csproj files.

Please review and let us know.

**Comment by daisukenishino** [comment]
@supragyan 

I have instructed the wrong instruction.
I was confused with this pull-request and following matters.
- https://github.com/OpenTouryoProject/OpenTouryoTemplates/issues/53
- https://github.com/OpenTouryoProject/OpenTouryo/issues/131#issuecomment-240626520

Is it possible to revert the following commit ?
- https://github.com/SymphonyTeleca/OpenTouryoTemplates/commit/c8a17e9
- https://github.com/SymphonyTeleca/OpenTouryoTemplates/commit/d87e538
- https://github.com/SymphonyTeleca/OpenTouryoTemplates/commit/58ff9c2

And please tell me about the corresponding situation of https://github.com/OpenTouryoProject/OpenTouryoTemplates/issues/53.

**Comment by daisukenishino** [comment]
@supragyan 

>And please tell me about the corresponding situation of https://github.com/OpenTouryoProject/OpenTouryoTemplates/issues/53.

It had been [completed](https://github.com/SymphonyTeleca/OpenTouryoTemplates/pull/170).

**Comment by supragyan** [comment]
@daisukenishino 

We have modified [this](https://github.com/SymphonyTeleca/OpenTouryoTemplates/pull/166/commits/c8a17e9b877a70e29b9987e4ecfe0ca8f1b53527) as per [166 issuecoment](https://github.com/SymphonyTeleca/OpenTouryoTemplates/pull/166#discussion_r77583226).

>Is it possible to revert the following commit ?

Are you surely want us to revert these above changes?

We will discuss on same on tomorrow meeting.

**Comment by daisukenishino** [comment]
@supragyan 

>Are you surely want us to revert these above changes?

Yes, please revert by using "git reset" command.

**Comment by supragyan** [comment]
@daisukenishino 

>Yes, please revert by using "git reset" command.

Thank you for confirmation.
We will do so.

**Comment by supragyan** [comment]
@daisukenishino 

We have reverted last three commits from [azuretmplt branch](https://github.com/SymphonyTeleca/OpenTouryoTemplates/pull/166/commits/c8a17e9b877a70e29b9987e4ecfe0ca8f1b53527) using Git Reset command.
So now these changes not present in azuretmplt branch.

But still we can see these above commit changes only from specific commit ids as follows:
- https://github.com/SymphonyTeleca/OpenTouryoTemplates/commit/c8a17e9
- https://github.com/SymphonyTeleca/OpenTouryoTemplates/commit/d87e538
- https://github.com/SymphonyTeleca/OpenTouryoTemplates/commit/58ff9c2

Is this ok or do we need to remove commit history from above links also?
Please confirm.

**Comment by daisukenishino** [comment]
@supragyan 

>Is this ok or do we need to remove commit history from above links also?
Please confirm.

However, these commits have been deleted from the following.
https://github.com/SymphonyTeleca/OpenTouryoTemplates/commits/azuretmplt-chngs

Also these commits have been deleted even the following.
image

Therefore, I thnk that there is no problem.

**Comment by supragyan** [comment]
@daisukenishino 

Thank you for your confirmation.

**Comment by daisukenishino** [comment]
@supragyan 

Because I copied the branch as follows, I will close this pull request.
- https://github.com/OpenTouryoProject/OpenTouryoTemplates/tree/azuretmplt-chngs
- https://github.com/OpenTouryoProject/OpenTouryoTemplates/compare/01-72...azuretmplt-chngs

**Pull request closed without merging**